conans/test/unittests/client/build/cpp_std_flags_test.py

```diff
@@ -1,195 +1,170 @@
+        self.assertEqual(_make_cppstd_flag("gcc", "4.8", "17"), None)
 
-        self.assertEqual(cppstd_flag("gcc", "5.1", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("gcc", "5.1", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("gcc", "5.1", "17"), '-std=c++1z')
+        self.assertEqual(_make_cppstd_flag("gcc", "4.9", "11"), '-std=c++11')
+        self.assertEqual(_make_cppstd_flag("gcc", "4.9", "14"), '-std=c++14')
+        self.assertEqual(_make_cppstd_flag("gcc", "4.9", "17"), None)
 
-        self.assertEqual(cppstd_flag("gcc", "7", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("gcc", "7", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("gcc", "7", "17"), '-std=c++17')
+        self.assertEqual(_make_cppstd_flag("gcc", "5", "11"), '-std=c++11')
+        self.assertEqual(_make_cppstd_flag("gcc", "5", "14"), '-std=c++14')
+        self.assertEqual(_make_cppstd_flag("gcc", "5", "gnu14"), '-std=gnu++14')
+        self.assertEqual(_make_cppstd_flag("gcc", "5", "17"), None)
 
-        self.assertEqual(cppstd_flag("gcc", "8", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("gcc", "8", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("gcc", "8", "17"), '-std=c++17')
-        self.assertEqual(cppstd_flag("gcc", "8", "20"), '-std=c++2a')
+        self.assertEqual(_make_cppstd_flag("gcc", "5.1", "11"), '-std=c++11')
+        self.assertEqual(_make_cppstd_flag("gcc", "5.1", "14"), '-std=c++14')
+        self.assertEqual(_make_cppstd_flag("gcc", "5.1", "17"), '-std=c++1z')
+
+        self.assertEqual(_make_cppstd_flag("gcc", "7", "11"), '-std=c++11')
+        self.assertEqual(_make_cppstd_flag("gcc", "7", "14"), '-std=c++14')
+        self.assertEqual(_make_cppstd_flag("gcc", "7", "17"), '-std=c++17')
+
+        self.assertEqual(_make_cppstd_flag("gcc", "8", "11"), '-std=c++11')
+        self.assertEqual(_make_cppstd_flag("gcc", "8", "14"), '-std=c++14')
+        self.assertEqual(_make_cppstd_flag("gcc", "8", "17"), '-std=c++17')
+        self.assertEqual(_make_cppstd_flag("gcc", "8", "20"), '-std=c++2a')
 
     def test_gcc_cppstd_defaults(self):
-        self.assertEqual(cppstd_default("gcc", "4"), "gnu98")
-        self.assertEqual(cppstd_default("gcc", "5"), "gnu98")
-        self.assertEqual(cppstd_default("gcc", "6"), "gnu14")
-        self.assertEqual(cppstd_default("gcc", "6.1"), "gnu14")
-        self.assertEqual(cppstd_default("gcc", "7.3"), "gnu14")
-        self.assertEqual(cppstd_default("gcc", "8.1"), "gnu14")
+        self.assertEqual(_make_cppstd_default("gcc", "5"), "gnu98")
+        self.assertEqual(_make_cppstd_default("gcc", "6"), "gnu14")
+        self.assertEqual(_make_cppstd_default("gcc", "6.1"), "gnu14")
+        self.assertEqual(_make_cppstd_default("gcc", "7.3"), "gnu14")
+        self.assertEqual(_make_cppstd_default("gcc", "8.1"), "gnu14")
 
     def test_clang_cppstd_flags(self):
-        self.assertEqual(cppstd_flag("clang", "2.0", "98"), None)
```

Comment:
I would prefer a 1-1 test modification, so it is easier to check what are the changes here. I would expect changing strictly the ``cppstd_flag()`` => ``_make_cppstd_flag()``, but the rest exactly the same. Why so many changes?

Reply:
now it's 1-1